Public/Invoke-DockerPull.ps1

```diff
@@ -1,28 +1,34 @@
 function Invoke-DockerPull {
     [CmdletBinding()]
     param (
+
+        [Parameter(ValueFromPipelineByPropertyName = $true)]
         [ValidateNotNullOrEmpty()]
         [String]
-        $Registry,
+        $Registry = $global:DockerPublicRegistry,
 
         # Pull by name, by name + tag, by name + digest
-        [Parameter(mandatory = $true,ParameterSetName = 'WithImageOnly')]
-        [Parameter(mandatory = $true,ParameterSetName = 'WithImageAndDigest')]
-        [Parameter(mandatory = $true,ParameterSetName = 'WithImageAndTag')]
+        [Parameter(mandatory = $true, ParameterSetName = 'WithImageOnly', ValueFromPipelineByPropertyName = $true)]
+        [Parameter(mandatory = $true, ParameterSetName = 'WithImageAndDigest', ValueFromPipelineByPropertyName = $true)]
+        [Parameter(mandatory = $true, ParameterSetName = 'WithImageAndTag', ValueFromPipelineByPropertyName = $true)]
         [String]
         $ImageName,
 
         [ValidateNotNullOrEmpty()]
-        [Parameter(mandatory = $true,ParameterSetName = 'WithImageAndTag')]
+        [Parameter(mandatory = $true, ParameterSetName = 'WithImageAndTag', ValueFromPipelineByPropertyName = $true)]
         [String]
         $Tag = 'latest',
 
         [ValidateNotNullOrEmpty()]
-        [Parameter(mandatory = $true,ParameterSetName = 'WithImageAndDigest')]
+        [Parameter(mandatory = $true, ParameterSetName = 'WithImageAndDigest')]
         [String]
         $Digest = ''
```

Comment:
Consider this feature deprecated in the future as a separate PR? It is not usable in the pipeline scenario though.